### PR TITLE
Create CI with GitHub actions

### LINKED ISSUE #33: Create CI with GitHub actions
Just initial CI job to do compilation and test execution.

### PR DESCRIPTION
**Pull request by relentless-pursuit**
Resolves #33

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+      uses: actions/checkout@v2
+
+    - name: Set up JDK 21
+      uses: actions/setup-java@v2
+      with:
+        distribution: 'adopt'
+        java-version: '21'
+
+    - name: Grant execute permission for setup script
+      run: chmod +x setup.sh
```
Comment (romani) [inline .github/workflows/ci.yml:20]: Please update setup.sh to have execution permission, and let's push . No reason to give permission before each execution.

**Review by romani — CHANGES_REQUESTED** [review]
Item

File: .github/workflows/ci.yml (hunk 1 above)
Comment (romani) [reply .github/workflows/ci.yml:20]: @relentless-pursuit , please remove this line and lets merge this PR.

Comment (romani) [reply .github/workflows/ci.yml:20]: we can comment out 
```
- name: Run setup script
      run: ./setup.sh
```
for now 
and uncomment it when we be ready execution of checktyle repo

Comment (relentless-pursuit) [reply .github/workflows/ci.yml]: done

**Review by romani — APPROVED** [review]

**Pull request merged**